src/Microsoft.AspNet.Authentication/DataHandler/Serializer/TicketSerializer.cs

```diff
@@ -69,27 +77,35 @@ public static AuthenticationTicket Read([NotNull] BinaryReader reader)
             }
             string authenticationScheme = reader.ReadString();
             int identityCount = reader.ReadInt32();
-            var identities = new ClaimsIdentity[identityCount];
-            for (int i = 0; i != identityCount; ++i)
+            ClaimsPrincipal principal = null;
+
+            // Negative values are used to signify null
+            if (identityCount > 0)
```

Comment:
I think this should be >= 0. Right now if you serialize a non-null ClaimsPrincipal with 0 identities, you will get back a null principal.